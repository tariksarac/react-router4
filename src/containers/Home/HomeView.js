import React, { Component } from 'react';

class HomeView extends Component {
    constructor(props){
        super(props);

        this.state = {
            showMessage: false
        };

        this.changeUser = this.changeUser.bind(this)
    }

    componentDidMount(){
        this.props.getApps()
    }

    changeUser(){
        this.setState({showMessage: !this.state.showMessage})
    }

    getAllApps(){
        return this.props.apps.map(function (app) {
            return (
                <div key={app.id}>
                    <p>{app.name}</p>
                    <p>{app.description}</p>
                </div>
            )
        })

    }
    render() {
        let apps = this.props.apps ? this.getAllApps() : [];
        return (
            <div>
                <h2>Homepage</h2>
                <button onClick={this.changeUser}>Change user</button>
                <div id="messeage">
                    { this.state.showMessage ?
                        <p>Home page with message</p> :
                        <p>Home page</p>
                    }
                </div>
                {apps}
            </div>
        );
    }
}

HomeView.propTypes = {};
HomeView.defaultProps = {};

export default HomeView;

