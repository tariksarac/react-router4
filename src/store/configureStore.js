import { compose, createStore, applyMiddleware } from 'redux'
import rootReducer from '../reducers'
import reduxThunk from 'redux-thunk';

export default function configureStore(preloadedState) {
    return createStore(
        rootReducer,
        preloadedState,
        // applyMiddleware(reduxThunk),
        compose(
            applyMiddleware(reduxThunk), // add your middlewares here
            /**
             * Conditionally add the Redux DevTools extension enhancer
             * if it is installed.
             */
            window.devToolsExtension ? window.devToolsExtension() : f => f
        )

    )
}