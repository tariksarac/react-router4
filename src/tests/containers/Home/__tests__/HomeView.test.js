import React from 'react';
import HomeView from '../../../../containers/Home/HomeView'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import {Provider} from 'react-redux'
import configureStore from 'redux-mock-store'


describe('Home page tests', () => {

    const initialState = {};
    const mockStore = configureStore();
    const getApps = jest.fn();
    let store,component;

    beforeEach(()=>{
        store = mockStore(initialState);
        component = mount( <Provider store={store}><HomeView getApps={getApps}/></Provider> );
    });

    it('renders  Home without crashing', () => {
        const home = renderer.create(<HomeView getApps={getApps}/>);
        let tree = home.toJSON();
        expect(tree).toMatchSnapshot()
    });

    it('change the message for admin when click', () => {
        expect(component.find('div[id="messeage"]').text()).toEqual('Home page');
        component.find('button').simulate('click');
        expect(component.find('div[id="messeage"]').text()).toEqual('Home page with message')

    })

})

