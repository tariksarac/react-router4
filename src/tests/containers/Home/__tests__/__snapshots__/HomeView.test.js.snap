// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page tests renders  Home without crashing 1`] = `
<div>
  <h2>
    Homepage
  </h2>
  <button
    onClick={[Function]}
  >
    Change user
  </button>
  <div
    id="messeage"
  >
    <p>
      Home page
    </p>
  </div>
</div>
`;
